### PR TITLE
Initial merging of the Docker and Code repos for the Gateway.

### PR DESCRIPTION
**Pull request by zmoore-usgs**

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+ENV ribbonConnectTimeout=6000
+ENV ribbonReadTimeout=60000
+ENV hystrixThreadTimeout=10000000
+ENV oauthClientId=client-id
+ENV oauthClientAccessTokenUri=https://example.gov/oauth/token
+ENV oauthClientAuthorizationUri=https://example.gov/oauth/authorize
+ENV oauthResourceTokenKeyUri=https://example.gov/oauth/token_key
+ENV oauthResourceId=resource-id
+ENV aqcuLoginUrl=https://localhost:8443/
+ENV HEALTHY_RESPONSE_CONTAINS='{"status":"UP"}'
```
Comment (isuftin) [inline Dockerfile]: Is this the same as the base container? If so, it should probably not be in here. Same as any other ENV thatmatches that of the base container

File: docker-compose.yml [hunk 2]
```diff
@@ -0,0 +1,58 @@
+---
+  version: '3.5'
```
Comment (isuftin) [inline docker-compose.yml]: Update to 3.7

File: .travis.yml [hunk 3]
```diff
@@ -0,0 +1,38 @@
+---
+  sudo: required
+
+  env:
+    global:
+      - DOCKER_COMPOSE_VERSION=1.22.0
+      - DOCKER_IMAGE=aqcu-tss-report
```
Comment (mhines-usgs) [inline .travis.yml]: 👀

File: Dockerfile [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+# See:
+# https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=911925
+# https://github.com/carlossg/docker-maven/issues/92
+# FROM maven:3-jdk-8-slim AS build
+
+COPY pom.xml /build/pom.xml
+WORKDIR /build
+
+#download all maven dependencies (this will only re-run if the pom has changed)
+RUN mvn -B dependency:go-offline
```
Comment (mhines-usgs) [inline Dockerfile]: ooh, should i add this to the tss report? is this the magic caching?

File: docker/config.env [hunk 5]
```diff
@@ -0,0 +1,51 @@
...
+srsReportServerList=https://localhost:7506
+extReportServerList=https://localhost:7507
+svpReportServerList=https://localhost:7509
+vdiReportServerList=https://localhost:7510
+lookupServiceServerList=https://localhost:7503
+aqcuLoginUrl=https://localhost:7498/timeseries
+keystoreLocation=/home/spring/localkeystore.pkcs12
+keystoreSSLKey=tomcat
+SPRING_CLOUD_CONFIG_ENABLED=false
+TOMCAT_CERT_PATH=/home/spring/tomcat-wildcard-dev.crt
```
Comment (isuftin) [inline docker/config.env]: I think this and a number of these are defined in the base container and the values match so they can be removed here

**Comment by isuftin** [comment]
@zmoore-usgs Added a few comments but this looks like it should work

File: Dockerfile (hunk 1 above)
Comment (zmoore-usgs) [reply Dockerfile]: :+1:

File: Dockerfile (hunk 4 above)
Comment (zmoore-usgs) [reply Dockerfile]: Yeah, that's what Carl had added to the base image

**Pull request merged**